connector/openshift/openshift_test.go

```diff
@@ -184,6 +184,57 @@ func TestCallbackIdentity(t *testing.T) {
 	expectEquals(t, identity.Groups[0], "users")
 }
 
+func TestQueryOAuth2Endpoints(t *testing.T) {
+	s := newTestServer(map[string]interface{}{
+		"/oauth/authorize": nil,
+		"/oauth/token": nil,
+	})
+	defer s.Close()
+
+	hostURL, err := url.Parse(s.URL)
+	expectNil(t, err)
+
+	_, err = http.NewRequest("GET", hostURL.String(), nil)
+	expectNil(t, err)
+
+	h, err := newHTTPClient(true, "", nil)
+	expectNil(t, err)
+
+	oc := openshiftConnector{apiURL: s.URL, httpClient: h, oauth2Config: &oauth2.Config{
+		Endpoint: oauth2.Endpoint{
+			AuthURL: s.URL + "/oauth/authorize", TokenURL: s.URL + "/oauth/token",
+		},
+	}}
+	err = oc.queryOAuth2Endpoints()
+	expectNil(t, err)
+}
+
+func TestQueryOauth2EndpointsFailure(t *testing.T) {
+	s := newTestServer(map[string]interface{}{
+		"/oauth/authorize": nil,
+		"/oauth/token": nil,
+	})
+	defer s.Close()
+	hostURL, err := url.Parse(s.URL)
+	expectNil(t, err)
+
+	_, err = http.NewRequest("GET", hostURL.String(), nil)
+	expectNil(t, err)
+
+	s.Close()
+
+	h, err := newHTTPClient(true, "", nil)
+	expectNil(t, err)
+
+	oc := openshiftConnector{apiURL: s.URL, httpClient: h, oauth2Config: &oauth2.Config{
+		Endpoint: oauth2.Endpoint{
+			AuthURL: s.URL + "/oauth/authorize", TokenURL: s.URL + "/oauth/token",
+		},
+	}}
+	err = oc.queryOAuth2Endpoints()
+	expectNotNil(t, err)
+}
```

Comment:
Do you really need the test server for this? It's already closed at this point. I think the test would be way more meaningful if you exercise both the success and the failure scenario. In the current form, you're unable to spot any systematic brokenness in `queryOAuth2Endpoints`

Reply:
Hm, I'm not sure I understood your comment w.r.t testing the success and failure scenarios.
Right now I _am_ testing the failure / success scenario by having the two tests where one handles success and one failure. 
Could you elaborate more what you mean, I think I'm not following properly tbh